### PR TITLE
Add possibility to configure custom search component

### PR DESCRIPTION
**Pull request by dsanko-sc**
#44

### CODE REVIEW

File: src/client/components/ReferenceSearchDialogProps/index.js [hunk 1]
```diff
@@ -31,7 +31,8 @@ export default {
   searchFields: PropTypes.arrayOf(PropTypes.shape({
     name: PropTypes.string.isRequired,
     label: PropTypes.string.isRequired,
-    inputComponent: PropTypes.func
+    inputComponent: PropTypes.func,
+    additionalProps: PropTypes.func
```
Comment (asergeev-sc) [inline src/client/components/ReferenceSearchDialogProps/index.js]: Not clear function purpose and its interface (input, output)

File: src/client/components/ReferenceSearchDialog/ReferenceSearchDialog.react.js [hunk 2]
```diff
@@ -338,10 +345,14 @@ export default class ReferenceSearchDialog extends Component {
...
                           className="form-control"
-                          onChange={(event) => this.onChangeSearchParam(
-                            column.name, event.target.value
+                          onChange={(event) => this.onChangeSearchParams(
+                            { [column.name]: event.target.value }
                           )}
                           value={searchParams[column.name] || ''}
+                          {...(additionalProps && additionalProps({
+                            searchParams,
+                            onChangeSearchParams: this.onChangeSearchParams
```
Comment (asergeev-sc) [inline src/client/components/ReferenceSearchDialog/ReferenceSearchDialog.react.js]: Not clear why 'searchParams' and 'onChangeSearchParams' are passed into onChangeSearchParams?

File: src/client/components/ReferenceSearchDialog/ReferenceSearchDialog.react.js [hunk 3]
```diff
@@ -118,14 +118,16 @@ export default class ReferenceSearchDialog extends Component {
     }
   }
 
-  onChangeSearchParam = (name, value) => {
+  onChangeSearchParams = (modifiedSearchParams) => {
```
Comment (asergeev-sc) [inline src/client/components/ReferenceSearchDialog/ReferenceSearchDialog.react.js]: Not clear why this change in internal API is even required?

**Comment by dsanko-sc** [comment]
Won't be merged.
